File: Algorithms/Searching Algorithms/Linear Search/JavaScript/LinearSearch.js
/**
 * @author Patryk Jóźwiak
 */
function linearSearch(haystack, needle) {
    var len = haystack.length;
    // Go through all values
    for (var i = 0; i < len; i++) {
        // Check if current value is equel to search value
        if (haystack[i] === needle) return i;
    }
    return -1;
}

var haystack = [1, 2, 3, 4, 9, 51, 54, 65, 45, 77];
console.log( linearSearch(haystack, 65) );

File: Algorithms/Searching Algorithms/Linear Search/Python/LinearSearch.py
class LinearSearch():
    def search(self, haystack, needle):
        length = len(haystack)
        # go through all values
        for i in range(length-1):
            # check if current value is search value
            if haystack[i] == needle:
                return i
        return -1 

ls = LinearSearch()
haystack = [2, 3, 5, 7, 11, 13, 17, 19]

print( ls.search(haystack, 7) )

